core/src/main/java/org/apache/iceberg/DataFiles.java

```diff
@@ -89,6 +89,31 @@ static PartitionData fillFromPath(PartitionSpec spec, String partitionPath, Part
     return data;
   }
 
+  static PartitionData fillFromValues(
+      PartitionSpec spec, List<String> partitionValues, PartitionData reuse) {
+    PartitionData data = reuse;
+    if (data == null) {
+      data = newPartitionData(spec);
+    }
+
+    Preconditions.checkArgument(
```

Comment:
Could just do one check here size != size, "Invalid partition data, found %d fields, expected %d"

Reply:
Sure, I'll just check for `partitionValues.size() == spec.fields().size()`. The code here was copied from `fillFromPath`, where there are two checks.
Btw, I think we need to use "%s" in the string template, as the guava code only looks for "%s", to replace with the varargs.